ui/ozone/platform/wayland/ozone_platform_wayland.cc

```diff
@@ -75,6 +75,10 @@ class OzonePlatformWayland : public OzonePlatform {
   }
 
   void QueryHostDisplaysData(QueryHostDisplaysDataCallback callback) override {
+    // Make sure display output data is available by processing all pending
+    // requests.
+    wl_display_roundtrip(connection_->display());
```

Comment:
"It is not allowed to call this function while the thread is being prepared for reading events,
 and doing so will cause a dead lock.". Maybe we better remove the previous patch we made and do the following to know the dimensions in advance?:

wl_display* display = wl_display_connect(nullptr);

static const struct wl_registry_listener registry_output = {
    OzonePlatformWayland::HandleDisplayGeometry,
};

wl_registry* registry = wl_display_get_registry(display);
wl_registry_add_listener(registry, &registry_output, this);

while(bounds_.empty()) {
  if (wl_display_roundtrip(display) < 0)
    return;
} 

// close connection and destroy display.
return bounds_;

and in OzonePlatformWayland::HandleDisplayGeometry() {
  if (strcmp(interface, "wl_output") == 0) 
    // get bounds here.
}

Reply:
done